pkg/estimator/client/cache.go

```diff
@@ -80,12 +80,12 @@ func (c *SchedulerEstimatorCache) GetClient(name string) (estimatorservice.Estim
 }
 
 // EstablishConnection establishes a new gRPC connection with the specified cluster scheduler estimator.
-func EstablishConnection(kubeClient kubernetes.Interface, name string, estimatorCache *SchedulerEstimatorCache, port int) error {
+func EstablishConnection(kubeClient kubernetes.Interface, name, namespace string, estimatorCache *SchedulerEstimatorCache, port int) error {
 	if estimatorCache.IsEstimatorExist(name) {
 		return nil
 	}
 
-	serverAddr, err := resolveCluster(kubeClient, util.NamespaceKarmadaSystem,
+	serverAddr, err := resolveCluster(kubeClient, namespace,
```

Comment:
I'm afraid this does not work. Our service is created in karmada-host, `resolveCluster` try to find a service in karmada control plane which is not karmada-host. So you may not able to find this service. And then it still uses `names.GenerateEstimatorServiceName(name)` as an address. So this flag does not work here.